### PR TITLE
Ewilliams/enable custom tags

### PR DESCRIPTION
**Pull request by EuanOBC**
Adds capability to use custom tags in database. Will require database in aivr-signal-vegetation-classifier-dataset to be updated before including this submodule (ill do once dust settled with alfies changes)

### CODE REVIEW

File: config.py.template [hunk 1]
```diff
@@ -27,6 +27,15 @@ LABEL_MAP = {
     LABEL_OK: "ok",
 }
 
+# Custom Tags - tags custom to each dataset which add data about an image, but are not labels.
+CUSTOM_TAG_OVERHEAD = # For example to denote an overhead camera.
```
Comment (theycallhimpat) [inline config.py.template:31]: make it clear that this is an example and will need renaming. Or change to CUSTOM_TAG_CHANGEME or CUSTOM_TAG_XXXXXX

File: export.py [hunk 2]
```diff
@@ -18,63 +18,105 @@
...
-            STATUS_KEEP,
-        )
+        cur.execute("SELECT * FROM processed_images WHERE status=? AND labels IS NOT NULL AND hard=0;", [STATUS_KEEP])
+
     results = list(cur.fetchall())
-    print(f"Found {len(results)} images to export for {export_name}")
+    return results
+
+
+def retrieve_image_label(row: Dict[str, Any], custom_tag_filter: Optional[List[str]] = None) -> Optional[str]:
```
Comment (theycallhimpat) [inline export.py:34]: function name doesn't really match function description and implementation?

File: export.py [hunk 3]
```diff
@@ -18,63 +18,105 @@
...
+    """Checks if we should export a single image."""
```
Comment (theycallhimpat) [inline export.py:35]: please update description to explain a bit more about what it does (i.e. what criteria does it use to decide)

File: export.py [hunk 4]
```diff
@@ -18,63 +18,105 @@
...
+    else:
+        dst_image = src_image
+
+    cv2.imwrite(dst_file, dst_image)
+
+    return resized
+
+
+def export_classifier(hard: bool = False, custom_tag_filter: Optional[List[str]] = None) -> None:
+    export_name = "classifier"
```
Comment (theycallhimpat) [inline export.py:96]: what about if we want test different classifiers from the same dataset? Should export_name be passed in as an argument (but could default to "classifier")

File: export.py [hunk 5]
```diff
@@ -18,63 +18,105 @@
...
-            if class_found:
-                break
-        else:
-            print_error(f"Unhandled image: {row}")
-            continue
-
-        if class_dir is None:
+        class_name = retrieve_image_label(row, custom_tag_filter)
+        if class_name is None:
             continue
```
Comment (theycallhimpat) [inline export.py:112]: maybe have a count of the number of images that were skipped?

File: config.py.template (hunk 1 above)
Comment (EuanOBC) [reply config.py.template]: 5f917a changed to changeme

File: export.py (hunk 2 above)
Comment (EuanOBC) [reply export.py]: afa57c Updated description

File: export.py (hunk 3 above)
Comment (EuanOBC) [reply export.py]: 99dccb Updated name

File: export.py (hunk 5 above)
Comment (EuanOBC) [reply export.py:114]: added counter in c8cfbe

**Pull request merged**